### PR TITLE
Trying to add icibmcloud GetDNSZone

### PR DESCRIPTION
**Pull request by 23TNC**
We may not need all three variables from Zone, and it may be a good idea to os.Setenv when the apikey is queried, I kept it with the GetDNSZone call that requires it

Signed-off-by: wolf <cory.klokman@ibm.com>

### CODE REVIEW

File: pkg/types/powervs/platform.go [hunk 1]
```diff
@@ -14,6 +14,9 @@ type Platform struct {
 	UserID string `json:"userid"`
 	APIKey string `json:"apikey"`
 
+	DNSDomainName   string `json:"dnsdomainname"`
+	CISInstanceCRN  string `json:"cisinstancecrn"`
+	ResourceGroupID string `json:"resourcegroupid"`
```
Comment (clnperez) [inline pkg/types/powervs/platform.go:19]: I see this in the ibmcloud code as well, but I don't actually see it ever used anywhere. Do you know why it's been added?

File: pkg/asset/installconfig/powervs/session.go [hunk 2]
```diff
@@ -23,6 +25,7 @@ var (
 type Session struct {
 	Session *ibmpisession.IBMPISession
 	Creds   *UserCredentials
+	Zone    *icibmcloud.Zone
```
Comment (clnperez) [inline pkg/asset/installconfig/powervs/session.go:28]: I think this Session object is turning into a catchall for extra variables. The Session was the PI Session before, which doesn't have anything to do with a DNS zone. I think we need to organize all this a bit more clearly. For now, though, we can leave it. But can you just rename Zone to DNSZone? There are a two different types zones in this part of the code, and the fact that the client only named the DNS zone `Zone` is a bit confusing in this context.

**Review by clnperez — COMMENTED** [review]
What is going to call GetSession() that needs to get the DNS Zone? Is the purpose of this PR to set up for the survey question to query and return zones?

**Comment by 23TNC** [comment]
Closing this as PR 60 replaces this and handles the rebase

**Pull request closed without merging**